core/src/operations/decorators/advice.rs

```diff
@@ -48,18 +48,29 @@ pub enum AdviceInjector {
     MerkleNodeToStack,
 
     /// Pushes a list of field elements onto the advice stack. The list is looked up in the advice
-    /// map using the top 4 elements (i.e. word) from the operand stack as the key.
+    /// map using the specified word from the operand stack as the key. If `include_len` is set to
+    /// true, the number of elements in the value is also pushed onto the advice stack.
     ///
     /// Inputs:
-    ///   Operand stack: [KEY, ...]
+    ///   Operand stack: [<key_offset>, KEY, ...]
```

Comment:
If I understand correctly the `key_offset` is never actually on the operand stack, it is an immediate value on the instruction?  If my understanding is correct then maybe we could consider an alternate way of representing this.

```
Inputs:
    Immediate operator value: key_offset ? key_offset : 0
    Operand Stack: [KEY, ...]
    ...
```

In the future we may want to consider a variant of the instruction that takes the offset from the top of the stack however I don't think it's a requirement at the minute.

Reply:
> If I understand correctly the `key_offset` is never actually on the operand stack, it is an immediate value on the instruction?

This is correct. The `<key_offset>` was meant to indicate that the `KEY` is not at the top of the stack (i.e., there is some number of elements on the stack which is above the `KEY`). But probably I could have done it just like so:
```
[..., KEY, ...]
```